src/App.tsx

```diff
@@ -15,48 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NAME = 'name',
+  LENGTH = 'length',
+  NONE = 'none',
+}
+
+function sortArray(
+  goods: string[],
+  howSort: SortType,
+  isReversed: boolean,
+) {
+  const array = [...goods];
+
+  array.sort((value1, value2) => {
+    switch (howSort) {
+      case SortType.NAME:
+        return value1.localeCompare(value2);
+
+      case SortType.LENGTH:
+        return value1.length - value2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    array.reverse();
+  }
+
+  return array;
+}
+
 export const App: React.FC = () => {
+  const [sort, setSort] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = sortArray(
+    goodsFromServer,
+    sort,
+    isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            { 'is-light': sort !== SortType.NAME },
+          )}
+          onClick={() => setSort(SortType.NAME)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button',
+            'is-success',
+            { 'is-light': sort !== SortType.LENGTH },
+          )}
+          onClick={() => setSort(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(prevIsReversed => !prevIsReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sort !== SortType.NONE || isReversed)
+          && (
+            <button
+              type="button"
+              className={classNames(
+                'button',
+                'is-danger',
+                { 'is-light': !(sort !== SortType.NONE || isReversed) },
+              )}
+              onClick={() => {
+                setSort(SortType.NONE);
+                setIsReversed(false);
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here